packages/turbo-codemod/src/transforms/migrate-dot-env.ts

```diff
@@ -0,0 +1,103 @@
+import path from "node:path";
+import { readJsonSync, existsSync } from "fs-extra";
+import { type PackageJson, getTurboConfigs } from "@turbo/utils";
+import type { Schema as TurboJsonSchema } from "@turbo/types";
+import type { Transformer, TransformerArgs } from "../types";
+import { getTransformerHelpers } from "../utils/getTransformerHelpers";
+import type { TransformerResults } from "../runner";
+
+// transformer details
+const TRANSFORMER = "migrate-dot-env";
+const DESCRIPTION = 'Migrate the "dotEnv" entries to "inputs" in `turbo.json`';
+const INTRODUCED_IN = "2.0.0";
+
+function migrateConfig(config: TurboJsonSchema) {
+  if ("globalDotEnv" in config) {
+    if (config.globalDotEnv) {
+      config.globalDependencies = config.globalDependencies ?? [];
+      for (const dotEnvPath of config.globalDotEnv) {
+        config.globalDependencies.push(dotEnvPath);
+      }
+    }
+    delete config.globalDotEnv;
+  }
+
+  for (const [_, taskDef] of Object.entries(config.pipeline)) {
+    if ("dotEnv" in taskDef) {
+      if (taskDef.dotEnv) {
+        taskDef.inputs = taskDef.inputs ?? ["$TURBO_DEFAULT$"];
```

Comment:
not blocking, but feels a bit muddy to include `$TURBO_DEFAULTS$` here 🤔 . I see we don't have a codemod to add this in though

Reply:
makes sense